scripts/calculateRevenue/protocols/utils/networks.ts

```diff
@@ -0,0 +1,49 @@
+import { QueryResponse, TransactionField } from '@envio-dev/hypersync-client'
+import { getHyperSyncClient } from '../../../utils'
+import { NetworkId } from '../../../types'
+
+export async function fetchTotalGasUsed({
+  networkId,
+  users,
+  startBlock,
+  endBlock,
+}: {
+  networkId: NetworkId
+  users: string[]
+  startBlock?: number
+  endBlock?: number
+}): Promise<number> {
+  let totalGasUsed = 0
+  let hasMoreBlocks = true
+
+  try {
+    const client = getHyperSyncClient(networkId)
+
+    const query = {
+      transactions: [{ from: users }],
+      fieldSelection: {
+        transaction: [TransactionField.GasUsed, TransactionField.GasPrice],
+      },
+      fromBlock: startBlock ?? 0,
+      ...(endBlock && { toBlock: endBlock }),
+    }
+
+    while (hasMoreBlocks) {
+      const response: QueryResponse = await client.get(query)
+      if (response.nextBlock === query.fromBlock) {
+        hasMoreBlocks = false
+      } else {
+        query.fromBlock = response.nextBlock
+      }
+
+      for (const tx of response.data.transactions) {
+        totalGasUsed += Number(tx.gasUsed ?? 0) * Number(tx.gasPrice ?? 0)
```

Comment:
Should these be BitInts?  I imagine that's why they come back as strings instead of numbers.

Reply:
[CalculateRevenueFn](http://github.com/divvi-xyz/divvi-protocol-v0/blob/5dbfc211271f64cb670199b6451b4f9eafb6c14a/scripts/types.ts#L27-L31) has the return type of `Promise<number>`, but maybe we should change this return type for all `CalculateRevenueFn` functions separate PR?

Reply:
OK, good suggestion.  Leave it as Number for now :+1: